### PR TITLE
Add an option for opening PRs with golden run result in reusable test workflow

### PR DESCRIPTION
**Pull request by fbeutin-ledger**

### CODE REVIEW

File: .github/workflows/_open_pr_with_new_snapshots.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            gh label create 'auto' --color 'b4a8d1' --description 'Automatically created'
+          fi
+
+          # Create the PR with a placeholder body. Will be consolidated at a later step
+          pr_url=$(gh pr create \
+                     --base "${{ inputs.app_branch_name }}" \
+                     --head "$branch_name" \
+                     --label "auto" \
+                     --title "[auto-update] Update Ragger snapshots" \
+                     --body "Created by a Github workflow")
```
Comment (tdejoigny-ledger) [inline .github/workflows/_open_pr_with_new_snapshots.yml:100]: `--body "Created by a Github workflow")`
Here, we clearly need to raise the developers' awareness of the importance of properly reviewing that the generated snapshots are the expected ones. Bonus: If we could add a small warning icon, that would be great ;-)

**Review by tdejoigny-ledger — COMMENTED** [review]
could be relevant to document this new feature in the repo readme

**Comment by fbeutin-ledger** [comment]
> could be relevant to document this new feature in the repo readme
Not sure of this, the readme is just the high level overview, the option is described in the workflow API alongside the other options

File: .github/workflows/_open_pr_with_new_snapshots.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+          fetch-depth: 0
+          # needed, else the push inside the action will use default credentials
+          # instead of provided ones
+          persist-credentials: false
+
+      - name: Download new snapshots
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ inputs.snapshots_artifact_name }}
+          path: ${{ inputs.snapshots_directory }}/snapshots
```
Comment (cedelavergne-ledger) [inline .github/workflows/_open_pr_with_new_snapshots.yml:55]: suffix /snapshots is already given in `reusable_ragger_tests.yml`?

File: .github/workflows/_open_pr_with_new_snapshots.yml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+          git add ${{ inputs.snapshots_directory }}/snapshots
+          git status
+          git commit -m "[auto-update] Update Ragger snapshots"
+
+          # Use the commit sha in the branch name to allow this job to be parallelized
+          id=$(git rev-parse --short HEAD)
+          echo "ID: $id"
+          branch_name="auto/update_snapshots-$id"
+          echo "Branch Name: $branch_name"
+          git checkout -b $branch_name
```
Comment (cedelavergne-ledger) [inline .github/workflows/_open_pr_with_new_snapshots.yml:85]: Isn't it better to switch to a new branch before committing?

**Review by tdejoigny-ledger — APPROVED** [review]

File: .github/workflows/_open_pr_with_new_snapshots.yml (hunk 2 above)
Comment (fbeutin-ledger) [reply .github/workflows/_open_pr_with_new_snapshots.yml:55]: Nice catch, that was a last minute change that I did not test

File: .github/workflows/_open_pr_with_new_snapshots.yml (hunk 3 above)
Comment (fbeutin-ledger) [reply .github/workflows/_open_pr_with_new_snapshots.yml:85]: Technically it's the same but I agree that the idea of branching first and commiting second is better
However I use the commit sha to create the branch so I can't reverse the order

**Review by cedelavergne-ledger — APPROVED** [review]

**Pull request merged**